src/audio/copier/copier.c

```diff
@@ -208,82 +208,61 @@ static int create_host(struct comp_dev *parent_dev, struct copier_data *cd,
 		       const struct ipc4_copier_module_cfg *copier_cfg,
 		       int dir)
 {
-	struct sof_uuid host = {0x8b9d100c, 0x6d78, 0x418f, {0x90, 0xa3, 0xe0,
-			0xe8, 0x05, 0xd0, 0x85, 0x2b}};
 	struct ipc_config_host ipc_host;
-	const struct comp_driver *drv;
-	struct comp_dev *dev;
 	struct host_data *hd;
 	int ret = 0;
-
-	drv = ipc4_get_drv((uint8_t *)&host);
-	if (!drv)
-		return -EINVAL;
+	enum sof_ipc_frame __sparse_cache in_frame_fmt, out_frame_fmt;
+	enum sof_ipc_frame __sparse_cache in_valid_fmt, out_valid_fmt;
 
 	config->type = SOF_COMP_HOST;
 
-	ret = create_endpoint_buffer(parent_dev, cd, config, copier_cfg, ipc4_gtw_host, false, 0);
-	if (ret < 0)
-		return ret;
+	audio_stream_fmt_conversion(copier_cfg->base.audio_fmt.depth,
+				    copier_cfg->base.audio_fmt.valid_bit_depth,
+				    &in_frame_fmt,
+				    &in_valid_fmt,
+				    copier_cfg->base.audio_fmt.s_type);
 
-	memset(&ipc_host, 0, sizeof(ipc_host));
-	ipc_host.direction = dir;
-	ipc_host.dma_buffer_size = copier_cfg->gtw_cfg.dma_buffer_size;
+	audio_stream_fmt_conversion(copier_cfg->out_fmt.depth,
+				    copier_cfg->out_fmt.valid_bit_depth,
+				    &out_frame_fmt,
+				    &out_valid_fmt,
+				    copier_cfg->out_fmt.s_type);
 
-	dev = comp_alloc(drv, sizeof(*dev));
-	if (!dev) {
-		ret = -EINVAL;
-		goto e_buf;
+	if (cd->direction == SOF_IPC_STREAM_PLAYBACK) {
+		if (config->type == SOF_COMP_HOST)
+			config->frame_fmt = in_frame_fmt;
+		else
+			config->frame_fmt = out_frame_fmt;
+	} else {
+		if (config->type == SOF_COMP_HOST)
+			config->frame_fmt = out_frame_fmt;
+		else
+			config->frame_fmt = in_frame_fmt;
 	}
```

Comment:
this is a `create_host()` function, can `config->type == SOF_COMP_HOST` be false here?

Reply:
seems not, later code need it to determine frame format

Reply:
@btian1 if (config->type == SOF_COMP_HOST) this will never be false. That means you only need to check for direction and set the confg_frame_fmt accordingly here. The check for SOF_COMP_HOST is irrelevant

Reply:
Thanks, @lyakh @ranj063 , will change accordingly.